presto-oracle/src/main/java/io/prestosql/plugin/oracle/OracleClientModule.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.oracle;
+
+import com.google.inject.Binder;
+import com.google.inject.Provides;
+import com.google.inject.Scopes;
+import com.google.inject.Singleton;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
+import io.prestosql.plugin.jdbc.BaseJdbcConfig;
+import io.prestosql.plugin.jdbc.ConnectionFactory;
+import io.prestosql.plugin.jdbc.DriverConnectionFactory;
+import io.prestosql.plugin.jdbc.JdbcClient;
+import io.prestosql.plugin.jdbc.credential.CredentialProvider;
+import oracle.jdbc.OracleDriver;
+
+import java.sql.SQLException;
+import java.util.Properties;
+
+import static io.airlift.configuration.ConfigBinder.configBinder;
+
+public class OracleClientModule
+        extends AbstractConfigurationAwareModule
+{
+    @Override
+    protected void setup(Binder binder)
+    {
+        binder.bind(JdbcClient.class).to(OracleClient.class).in(Scopes.SINGLETON);
+        configBinder(binder).bindConfig(BaseJdbcConfig.class);
+        configBinder(binder).bindConfig(OracleConfig.class);
+    }
+
+    @Provides
+    @Singleton
+    public static ConnectionFactory connectionFactory(BaseJdbcConfig config, CredentialProvider credentialProvider, OracleConfig oracleConfig)
+            throws SQLException
+    {
+        Properties connectionProperties = new Properties();
+        connectionProperties.setProperty("defaultRowPrefetch", String.valueOf(oracleConfig.getDefaultRowPrefetch()));
+        connectionProperties.setProperty("defaultBatchValue", String.valueOf(oracleConfig.getDefaultBatchValue()));
```

Comment:
i don't think it's valuable, since we handle batching explicitly. Please correct me if am wrong.

Reply:
we put this to allow configuration.. if this are handle explicitly we can remove it

Reply:
This property is for "Oracle Update Batching" which [no longer exists](https://docs.oracle.com/en/database/oracle/oracle-database/19/jjdbc/performance-extensions.html#GUID-20BF8463-A574-48CB-AD56-18CF8DEF603D):

> Oracle update batching was deprecated in Oracle Database 12c Release 1 (12.1). Starting in Oracle Database 12c Release 2 (12.2), Oracle update batching is a no operation code (no-op). This means that if you implement Oracle update batching in your application, using the Oracle Database 12c Release 2 (12.2) JDBC driver, then the specified batch size is not set and results in a batch size of 1. With this batch setting, your application processes one row at a time. Oracle strongly recommends that you use the standard JDBC batching if you are using the Oracle Database 12c Release 2 (12.2) JDBC driver.

You can read more about what this property did in the [earlier documentation](https://docs.oracle.com/database/121/JJDBC/oraperf.htm#JJDBC28757).

As @findepi mentions, we already implement standard JDBC batching in `JdbcPageSink`.

Thus, we should remove this property and the config, since it isn't used.

Reply:
config removed